mitosheet/mitosheet/api/graph/graph_utils.py

```diff
@@ -102,58 +85,101 @@ def get_graph_title(
+        )
+        y_axis_title = (
+            str(y_axis_column_headers[0]) if len(y_axis_column_headers) == 1 else ""
+        )
 
+    return x_axis_title, y_axis_title
 
-def create_parameter(param_key: str, param_value: str, format_as_string: bool) -> str:
-    """
-    Returns the params ready to be used in the transpiled code.
+
+def get_html_and_script_from_figure(
+    fig: go.Figure, height: int, width: int
+) -> Dict[str, str]:
     """
-    if format_as_string:
-        return f"{param_key}='{param_value}'"
-    else:
-        return f"{param_key}={param_value}"
+    Given a plotly figure, generates HTML from it, and returns
+    a dictonary with the div and script for the frontend.
 
+    The plotly HTML generated by the write_html function call is a div with two children:
+    1. a div that contains the id for the graph itself
+    2. a script that actually builds the graph
 
-def filter_df_to_top_unique_values_in_series(
-    df: pd.DataFrame,
-    main_series: pd.Series,
-    num_unique_values: int,
-) -> pd.Series:
+    Because we have to dynamically execute the script, we split these into two
+    strings, to make them easier to do what we need on the frontend
     """
-    Helper function for filtering the dataframe down to the top most common
-    num_unique_values in the main_series. Will not change the series if there are less
-    values than that.
+    # Send the graph back to the frontend
+    buffer = io.StringIO()
+    fig.write_html(
+        buffer,
+        full_html=False,
+        include_plotlyjs=False,
+        default_height=height,
+        default_width=width,
+    )
 
-    The function filters the entire dataframe to make sure that the columns stay
-    the same length (which is necessary if you want to graph them).
+    original_html = buffer.getvalue()
+    # First, we remove the main div, and the resulting whitespace, to just have the children
+    original_html = original_html[5:]
+    original_html = original_html[:-6]
+    original_html = original_html.strip()
 
-    It returns the filtered dataframe
-    """
-    if (
-        len(main_series) < num_unique_values
-        or main_series.nunique() < num_unique_values
-    ):
-        return df
+    # Then, we split the children into the div, and the script
+    # making sure to remove the script tag (so we can execute it)
+    script_start = '<script type="text/javascript">'
+    script_end = "</script>"
+    split_html = original_html.split(script_start)
+    div = split_html[0]
+    script = split_html[1][: -len(script_end)]
 
-    value_counts_series = main_series.value_counts()
-    most_frequent_values_list = value_counts_series.head(
-        n=num_unique_values
-    ).index.tolist()
+    return {"html": div, "script": script}
 
-    return df[main_series.isin(most_frequent_values_list)]
+
+"""
+PRESERVED FOR BACKWARDS COMPATABILITY. 
+This function is exported to users in the generated code. We will preserve it here so that 
```

Comment:
Nice! I was coming here to comment this, but good job recognizing it safely. 

Check out line 93 in `deprecated_utils.py` - it shows how you can wrap a function so that it gives a warning it's used externally!

We should do this for the filtering function as we're depreciating it!